src/axes/categoryAxis.ts

```diff
@@ -117,22 +130,100 @@ namespace Plottable.Axes {
     }
 
     protected _getTickValues() {
-      return this.getDownsampleInfo().domain;
+      return this._scale.getDownsampleInfo().domain;
     }
 
     /**
-     * Take the scale and drop ticks at regular intervals such that the resultant ticks are all a reasonable minimum
-     * distance apart. Return the resultant ticks to render, as well as the new stepWidth between them.
+     * Gets the current tick mark position value.
+     */
+    public tickMarkPosition(): ITickMarkPosition;
+    /**
+     * Sets  whether tick marks are drawn in the center of category or after it.
      *
-     * @param {Scales.Category} scale - The scale being downsampled. Defaults to this Axis' scale.
-     * @return {DownsampleInfo} an object holding the resultant domain and new stepWidth.
+     * @param {ITickMarkPosition} tickMarkPosition - "center" or "after"
+     * @returns {Category} The calling Category.
      */
-    public getDownsampleInfo(scale = <Scales.Category> this._scale): DownsampleInfo {
-      const downsampleRatio = Math.ceil(Category._MINIMUM_WIDTH_PER_LABEL_PX / scale.stepWidth());
-      return {
-        domain: scale.domain().filter((d, i) => i % downsampleRatio === 0),
-        stepWidth: downsampleRatio * scale.stepWidth(),
-      };
+    public tickMarkPosition(tickMarkPosition: ITickMarkPosition): this;
+    public tickMarkPosition(tickMarkPosition?: ITickMarkPosition): any {
+      if (arguments.length === 0) {
+        return this._tickMarkPosition;
+      } else {
+        this._tickMarkPosition = tickMarkPosition;
+      }
+      return this;
+    }
+
+    /**
+     * Gets the current alignment. If not set, null will be returned.
+     */
+    public tickTextAlignment(): string;
+    /**
+     * Set alignment of tick labels. Must be one of "left", "right",
+     * or "center", or a falsy value. Setting to a falsy value will
+     * reset tick label alignment to the default alignment.
+     *
+     * Note that alignment is applied before rotation (see "{@link
+     * tickLabelAngle}").
+     *
+     * @param {String} tickTextAlignment One of "left", "right", "center",
+     * or a falsy value.
+     *
+     * @returns {Category} The calling Category.
+     */
+    public tickTextAlignment(tickTextAlignment: string): this;
+    public tickTextAlignment(tickTextAlignment?: string): any {
```

Comment:
nit: accept a string literal union instead of a generic string. In fact we already have http://plottablejs.org/docs/classes/plottable.components.alignment.html